### PR TITLE
#750 Store test results in gcloud bucket

### LINKED ISSUE #750: Automatically store reports in gcloud bucket
It would be nice to automatically store the results in the gcloud bucket.

Specifically the html report and the junit xml report.

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #750 

-Should upload reports to gcloud bucket

## Checklist

- [X] Documented
- [X] Unit tested
- [X] release_notes.md updated
- [X] update README.md
- [X] update flank.yml
- [X] update flank.ios.yml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,6 +204,9 @@ flank:
 
   ## Enable create additional local junit result on local storage with failure nodes on passed flaky tests.
   # full-junit-result: false
+
+  ## Disables flank results upload on gcloud storage.
+  # disable-results-upload: false
```
[the remark this answers is not in the document]
Comment (pawelpasterz) [reply README.md:209]: And add info about default value, thanks!

File: test_runner/src/main/kotlin/ftl/reports/CostReport.kt [hunk 2]
```diff
@@ -41,5 +42,6 @@ object CostReport : IReport {
         val output = generate(matrices)
         if (printToStdout) print(output)
         write(matrices, output, args)
+        GcStorage.uploadReportResult(output, args, fileName())
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/reports/CostReport.kt:45]: Just want to confirm, we want to upload all reports or just junit's?

File: test_runner/src/main/kotlin/ftl/reports/HtmlErrorReport.kt [hunk 3]
```diff
@@ -28,10 +29,11 @@ object HtmlErrorReport : IReport {
...
     ) {
         val suites = result?.testsuites?.process()?.takeIf { it.isNotEmpty() } ?: return
-        Files.write(
-            Paths.get(reportPath(matrices, args)),
-            suites.createHtmlReport().toByteArray()
-        )
+        Paths.get(reportPath(matrices, args)).let {
+            val htmlReport = suites.createHtmlReport()
+            Files.write(it, htmlReport.toByteArray())
+            GcStorage.uploadReportResult(htmlReport, args, it.fileName.toString())
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/reports/HtmlErrorReport.kt:35]: Same here, do we want to have this uploaded?

File: test_runner/src/main/kotlin/ftl/reports/CostReport.kt (hunk 2 above)
Comment (bootstraponline) [reply test_runner/src/main/kotlin/ftl/reports/CostReport.kt:45]: I think we want to upload all reports.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamfilipow92) [reply README.md:210]: done

**Pull request merged**